core/connection/manager.go

```diff
@@ -198,8 +198,8 @@ func (manager *connectionManager) Connect(consumerID, accountantID identity.Iden
 	manager.setStatus(statusConnecting())
 	defer func() {
 		if err != nil {
-			manager.setStatus(statusNotConnected())
-			manager.publishStateEvent(NotConnected)
+			log.Err(err).Msg("Connect failed, disconnecting")
+			err = manager.Disconnect()
```

Comment:
It will produce bunch of false statuses like `Disconnecting`

Reply:
Does it really that matter? We could introduce another state "Cancelling" if we really gonna analyze and have state statistics..

Reply:
Yes it matters, as we dont want strange transitions: `Connecting -> Disconnecting`

UI blinks to state Disconnecting already too much. Disconnecting transition should happen just in flow:
Connected -> Disconnecting